azkaban-common/src/main/java/azkaban/version/VersionSetLoader.java

```diff
@@ -17,21 +17,37 @@
 package azkaban.version;
 
 import java.io.IOException;
+import java.util.List;
 
 public interface VersionSetLoader {
 
     /**
-     * @return versionSetNum after created post insert
+     * @return versionSetNum after created post insert.
      */
     int insertAndGetVersionSetNum(String versionSetJsonString, String versionSetMd5Hex) throws IOException;
 
     /**
-     * @return true if successful, otherwise false
+     * @return true if successful, otherwise false.
      */
     boolean removeVersionSet(String versionSetMd5Hex) throws IOException;
 
     /**
-     * @return versionSetNum corresponding to versionSetJsonString and versionSetMd5Hex
+     * @return versionSetNum corresponding to versionSetJsonString and versionSetMd5Hex.
      */
     int getVersionSetNum(String versionSetJsonString, String versionSetMd5Hex) throws IOException;
+
+    /**
+     * @return {@link VersionSet} corresponding to versionSetMd5Hex.
+     */
+    VersionSet getVersionSet(String versionSetMd5Hex) throws IOException;
+
+    /**
+     * @return {@link VersionSet} corresponding to versionSetNum.
+     */
+    VersionSet getVersionSet(int versionSetNum) throws IOException;
```

Comment:
Can you please rename this method to getVersionSetById?

Reply:
Done